lib/mix/tasks/hex.build.ex

```diff
@@ -190,6 +190,20 @@ defmodule Mix.Tasks.Hex.Build do
       |> Enum.join("\n")
       |> Mix.raise()
     end
+
+    if organization in [nil, "hexpm"] do
+      licenses_valid_or_warn(meta.licenses)
+    end
+  end
+
+  defp licenses_valid_or_warn([]), do: Hex.Shell.warn("\nYou have not included any licenses\n")
+
+  defp licenses_valid_or_warn(licenses) do
+    unless Enum.all?(licenses, fn l -> l in valid_license_ids() end) do
+      Hex.Shell.warn(
+        "\nYou have chosen 1 or more licenses that are not recognized by SPDX\nConsider using a license from https://spdx.org/licenses/\n"
+      )
+    end
```

Comment:
what do you think about showing which ones are not valid, perhaps something like this?

```suggestion
    invalid_licenses = licenses -- valid_license_ids()

    if invalid_licenses != [] do
      message = [
        "The following licenses are not recognized by SPDX.\n",
        Enum.map(invalid_licenses, &" * #{&1}\n"),
        "\nConsider using licenses from https://spdx.org/licenses"
      ]

      Hex.Shell.warn(message)
    end
```